### PR TITLE
mergin with experemental branch

### PR DESCRIPTION
**Pull request by AntsiferauGodel**
A lot of changes for better codestyle, separated one Manager into 3. moved command local variables in methods. Too many changes for one PR, my bad.

### CODE REVIEW

File: Unit-03/lab-file-storage/ConsoleFileStorage/ConsoleFileStorage/CommandPattern/Commands/ExceptionCommands/NotFoundCommand.cs [hunk 1]
```diff
@@ -1,38 +1,25 @@
...
         {
-            StringBuilder output = new StringBuilder();
-            foreach (string item in CommandArguments)
-            {
-                output.Append(item);
-                output.Append(" "); //readable format
-            }
-            MetaFileManager.Output.DisplayInformation(string.Format("Command <{0}> not found!", output));
+            const string message = "Command  not found!";
+            Manager.Output.DisplayInformation(message);
```
[the remark this answers is not in the document]
Comment (AntsiferauGodel) [reply Unit-03/lab-file-storage/ConsoleFileStorage/ConsoleFileStorage/CommandPattern/Commands/ExceptionCommands/NotFoundCommand.cs]: isn't const behaves like static field? i mean it is as same as passing it through args?

File: Unit-03/lab-file-storage/ConsoleFileStorage/ConsoleFileStorage/MetaFiles/EXP/MetaManager.cs [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+            {
+                LogIn(userName);
+            }
+            catch (Exception)
+            {
+                if (ConfigurationManager.AppSettings.Get(userName) != null)
+                {
+                    CreateUser(userName);
+                    LogIn(userName);
+                }
```
[the remark this answers is not in the document]
Comment (AntsiferauGodel) [reply Unit-03/lab-file-storage/ConsoleFileStorage/ConsoleFileStorage/MetaFiles/EXP/MetaManager.cs]: i'm trying to catch InvalidOperationException in first try block, which means "No element satisfies the condition in predicate" and applying "counter measures" (creating user in dictionary) so there cant be another exception in LogIn in catch block, i guess

File: Unit-03/lab-file-storage/ConsoleFileStorage/ConsoleFileStorage/MetaFiles/EXP/MetaManager.cs [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+            }
+            else
+            {
+                throw new DamagedFileException();
+            }
+        }
+        public void ExportToJson(string path)
+        {
+            FileStream fs = new FileStream(path, FileMode.Create);
+            using (fs)
```
[the remark this answers is not in the document]
Comment (AntsiferauGodel) [reply Unit-03/lab-file-storage/ConsoleFileStorage/ConsoleFileStorage/MetaFiles/EXP/MetaManager.cs]: moved

File: Unit-03/lab-file-storage/ConsoleFileStorage/ConsoleFileStorage/PromptInput/GetCommandLoop.cs [hunk 4]
```diff
@@ -1,46 +1,38 @@
...
+        private readonly CommandFactory commandFactory = new CommandFactory();
 
-        public GetCommandLoop(string userName,bool isNeedToInitialize = false)
+        public GetCommandLoop(string userName, bool isNeedToInitialize = false)
         {
-
-                metaFileManager = new MetaFiles.MetaFileManager(userName, isNeedToInitialize);
-            
-            
+                manager = new MetaFiles.EXP.Manager(userName, isNeedToInitialize);
```
[the remark this answers is not in the document]
Comment (AntsiferauGodel) [reply Unit-03/lab-file-storage/ConsoleFileStorage/ConsoleFileStorage/PromptInput/GetCommandLoop.cs]: done

File: Unit-03/lab-file-storage/ConsoleFileStorage/ConsoleFileStorage/MetaFiles/EXP/MetaManager.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (AntsiferauGodel) [reply Unit-03/lab-file-storage/ConsoleFileStorage/ConsoleFileStorage/MetaFiles/EXP/MetaManager.cs]: done

File: Unit-03/lab-file-storage/ConsoleFileStorage/ConsoleFileStorage/CommandPattern/Commands/FileCommands/FileRemoveCommand.cs [hunk 5]
```diff
@@ -1,33 +1,29 @@
...
             }
-            else
-            {
-                throw new InvalidNumerOfArgumentsException();
-            }
-            MetaFileManager.RemoveFile(fileName);
-            MetaFileManager.Output.DisplayInformation(string.Format("File {0} has been removed", fileName));
+            string fileName;
+            fileName = СommandArguments.Arguments[0];
+            Manager.RemoveFile(fileName);
```
[the remark this answers is not in the document]
Comment (AntsiferauGodel) [reply Unit-03/lab-file-storage/ConsoleFileStorage/ConsoleFileStorage/CommandPattern/Commands/FileCommands/FileRemoveCommand.cs]: done, thx

File: Unit-03/lab-file-storage/ConsoleFileStorage/ConsoleFileStorage/CommandPattern/Commands/FileCommands/FileUploadCommand.cs [hunk 6]
```diff
@@ -1,32 +1,26 @@
...
             {
-                throw new InvalidNumerOfArgumentsException();
+                throw new InvalidNumberOfArgumentsException();
             }
-            MetaFileManager.UploadFile(pathTofile);
-            MetaFileManager.Output.DisplayInformation("File was uploaded!");
-            
+            string pathTofile;
+            pathTofile = СommandArguments.Arguments[0];
+            Manager.UploadFile(pathTofile);
```
[the remark this answers is not in the document]
Comment (AntsiferauGodel) [reply Unit-03/lab-file-storage/ConsoleFileStorage/ConsoleFileStorage/CommandPattern/Commands/FileCommands/FileUploadCommand.cs]: done, thx

File: Unit-03/lab-file-storage/ConsoleFileStorage/ConsoleFileStorage/CommandPattern/Commands/UserCommands/UserInfoCommand.cs [hunk 7]
```diff
@@ -1,29 +1,29 @@
...
+            if (СommandArguments.Arguments.Count != 0)
             {
-                throw new InvalidNumerOfArgumentsException();
+                throw new InvalidNumberOfArgumentsException();
             }
-            output = MetaFileManager.UserInfo();
-            MetaFileManager.Output.DisplayInformation("User info: ");
+            IList<string> output;
+            output = Manager.UserInfo();
+            Manager.Output.DisplayInformation("User info: ");
```
[the remark this answers is not in the document]
Comment (AntsiferauGodel) [reply Unit-03/lab-file-storage/ConsoleFileStorage/ConsoleFileStorage/CommandPattern/Commands/UserCommands/UserInfoCommand.cs]: done, thx

File: Unit-03/lab-file-storage/ConsoleFileStorage/ConsoleFileStorage/CommandPattern/Exceptions/InvalidTypeOfFlagException.cs [hunk 8]
```diff
@@ -1,18 +1,19 @@
...
-using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
+using System.Runtime.Serialization;
 
 namespace ConsoleFileStorage.CommandPattern.Exceptions
 {
-    class InvalidTypeOfFlagException : Exception
+    [Serializable]
+    public class InvalidTypeOfFlagException : Exception
```
[the remark this answers is not in the document]
Comment (AntsiferauGodel) [reply Unit-03/lab-file-storage/ConsoleFileStorage/ConsoleFileStorage/CommandPattern/Exceptions/InvalidTypeOfFlagException.cs]: now i'm using it in export command

File: Unit-03/lab-file-storage/ConsoleFileStorage/ConsoleFileStorage/CommandPattern/Factories/CommandFactory.cs [hunk 9]
```diff
@@ -27,41 +28,25 @@ public CommandFactory()
...
-            }
-            else
+            ICliCommand command;
+            command = new NotFoundCommand { Manager = manager };
+            var result = cliCommands.FirstOrDefault(cmd => cmd.CommandName == commandArguments.CommandName);
+            if (result != null)
             {
-                return new NotFoundCommand() { CommandArguments = arguments, MetaFileManager = metaFileManager };
+                command = result;
+                command.СommandArguments = commandArguments;
```
[the remark this answers is not in the document]
Comment (AntsiferauGodel) [reply Unit-03/lab-file-storage/ConsoleFileStorage/ConsoleFileStorage/CommandPattern/Factories/CommandFactory.cs]: fixed, thx a lot)

File: Unit-03/lab-file-storage/ConsoleFileStorage/ConsoleFileStorage/MetaFiles/EXP/Extensions/MetaManagerExtensions.cs [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+            using (FileStream fs = new FileStream(filePath, FileMode.Open))
+            {
+                output = SHA256.Create().ComputeHash(fs);
+            }
+            return output;
+        }
+        public static string ConvertDataToIso8601(this MetaManager metaFileManager, DateTime dateTime)
+        {
+            string s_output = dateTime.ToString("yyyy-MM-dd");
+            return s_output;
```
[the remark this answers is not in the document]
Comment (AntsiferauGodel) [reply Unit-03/lab-file-storage/ConsoleFileStorage/ConsoleFileStorage/MetaFiles/EXP/Extensions/MetaManagerExtensions.cs]: fixed, thx)

File: Unit-03/lab-file-storage/ConsoleFileStorage/ConsoleFileStorage/MetaFiles/EXP/Extensions/MetaManagerExtensions.cs [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+        }
+        public static MetaFileList ConvertDictionaryToList(this MetaManager metaFileManager)
+        {
+            MetaFileList metaFileList = new MetaFileList();
+            var fileRepresents = metaFileManager.MetaFile.FileRepresents.Values.GetEnumerator();
+            while (fileRepresents.MoveNext())
+            {
+                metaFileList.Files.Add(fileRepresents.Current);
+            }
+            return metaFileList;
```
[the remark this answers is not in the document]
Comment (AntsiferauGodel) [reply Unit-03/lab-file-storage/ConsoleFileStorage/ConsoleFileStorage/MetaFiles/EXP/Extensions/MetaManagerExtensions.cs]: private set is preventing us from changing ref, not insides of list, and thx for addrange advise xD

File: Unit-03/lab-file-storage/ConsoleFileStorage/ConsoleFileStorage/MetaFiles/EXP/MetaManager.cs [hunk 12]
```diff
@@ -79,11 +80,19 @@ private void CreateUser(string userName)
...
-                currentUser = MetaFile.UserRepresents.First(x => x.Key == userName).Value;
+                currentUser = MetaFile.UserRepresents.FirstOrDefault(x => x.Key == userName).Value;
+                if(currentUser == default(UserRepresent))
+                {
+                    return false;
+                }
+                else
+                {
+                    return true;
+                }
```
Comment (HannaKeryna) [inline Unit-03/lab-file-storage/ConsoleFileStorage/ConsoleFileStorage/MetaFiles/EXP/MetaManager.cs]: MetaFile.UserRepresents.FirstOrDefault(x => x.Key == userName).Value - what happens if value with userName key is not presented? looks like null exception

Also return currentUser == default(UserRepresent) - looks better

**Pull request merged**